### PR TITLE
File output support

### PR DESCRIPTION
**Pull request by bhrg3se**
File output support like [elasticsearch-dump](https://github.com/elasticsearch-dump/elasticsearch-dump)

### CODE REVIEW

File: esdump/esdump.go [hunk 1]
```diff
@@ -98,6 +114,13 @@ func ElasticDump(opt Options) (err error) {
 	g.Go(func() error {
 		defer close(savedHits)
 
+		if isOutputFile {
+			if err2 := setDataFile(ctx, hits, savedHits, outputFile); err2 != nil {
+				return err2
+			}
+			return nil
```
Comment (tsaikd) [inline esdump/esdump.go]:
```suggestion
			return setDataFile(ctx, hits, savedHits, outputFile)
```

File: esdump/esdump.go [hunk 2]
```diff
@@ -143,16 +166,19 @@ func ElasticDump(opt Options) (err error) {
 	return
 }
 
-func parseElasticURL(esurl string) (entrypoint string, indexName string, err error) {
+func parseElasticURL(esurl string) (entrypoint string, indexName string, isFile bool, err error) {
 	u, err := url.Parse(esurl)
 	if err != nil {
 		return
 	}
+	if u.Scheme == "" {
```
Comment (tsaikd) [inline esdump/esdump.go:171]: How about using file URI scheme?
https://en.wikipedia.org/wiki/File_URI_scheme

File: esdump/esdump.go [hunk 3]
```diff
@@ -188,6 +214,43 @@ func getData(
...
 }
 
+func setDataFile(
+	ctx context.Context,
+	hits <-chan elasticMessage,
+	savedHits chan<- elasticMessage,
+	outputFile *os.File,
+) (err error) {
+	defer func() {
+		logger.Debug("setDataFile defer with err: ", err)
```
Comment (tsaikd) [inline esdump/esdump.go]:
```suggestion
		if err == nil {
			logger.Debug("setDataFile finish")
		} else {
			logger.Debug("setDataFile err: ", err)
		}
```

File: esdump/esdump.go [hunk 4]
```diff
@@ -188,6 +214,43 @@ func getData(
...
+
+		// Check if we need to terminate early
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			savedHits <- hit
+		}
+	}
+	logger.Debug("setDataFile finish")
```
Comment (tsaikd) [inline esdump/esdump.go]:
```suggestion
```

File: esdump/esdump.go [hunk 5]
```diff
@@ -188,6 +214,43 @@ func getData(
...
+		outputFile.WriteString("\n")
+
+		// Check if we need to terminate early
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			savedHits <- hit
+		}
+	}
```
Comment (tsaikd) [inline esdump/esdump.go]:
```suggestion
	encoder := json.NewEncoder(outputFile)
	for {
		select {
		case <-ctx.Done():
			return ctx.Err()
		case hit, ok := <-hits:
			if !ok {
				return nil
			}

			index := hit.Index

			logger.Debugf("setDataFile index:%q type:%q id:%q", index, hit.Type, hit.Id)

			if err = encoder.Encode(hit); err != nil {
				return err
			}
			if _, err = outputFile.WriteString("\n"); err != nil {
				return err
			}

			savedHits <- hit
		}
	}
```

**Review by tsaikd — CHANGES_REQUESTED** [review]

File: esdump/esdump.go (hunk 2 above)
Comment (bhrg3se) [reply esdump/esdump.go:171]: How about both file URI scheme or blank scheme for file.

Something like
`if u.Scheme == ""  || u.Scheme == "file" {`

**Pull request merged**